Slim/Music/Info.pm

```diff
@@ -476,8 +476,11 @@ sub setRemoteMetadata {
 		$currentBitrates{$url} = $track->prettyBitRate;
 	}
 	
-	if ( $meta->{cover} && $url =~ m|^http| ) {
-		Slim::Utils::Cache->new->set("remote_image_$url", $meta->{cover}, '30 days');
+	if ( $meta->{cover} ) {
+		my $handler = Slim::Player::ProtocolHandlers->handlerForURL($url);
+		if ( $handler && $handler->isRemote ) {
+			Slim::Utils::Cache->new->set("remote_image_$url", $meta->{cover}, '30 days');
+		}	
 	}
```

Comment:
We've discussed this before, and I can't find the discussion any more... what was the reason why you didn't want to do this caching in the podcast plugin's URL wrapper? IIRC you didn't like my suggestion to pre-cache these `remote_image_$url` values while parsing the podcasts because you thought that would be caching too much?

With this current change we'd cache the value for any track in a TIDAL or Spotify or whatever playlist when you start playing it. This often can be dozens or hundreds of entries, too, which otherwise would not be cached at all. 

Unless I'm completely wrong again, I'd still keep this local to the podcast plugin, and not touch this particular method.

Reply:
You're right about the amount of cache. Now I can't just store image in the wrapper, the reason is in the header of the podcast new PH'
```
When a $url from an OPML item is added to current playlist, a $track object is created 
and its image is obtained by calling handlerForUrl($url)->getMetadataFor($url). 
If this protocol handler is HTTP, it needs to find the image just by using this $url, 
there is no other information. XMLBrowser(s), when adding the url to the playlist, 
call setRemoteMetadata which caches the OPML items's image link as "remote_image_$url". 
Then the getMetadataFor uses this as one of the image sources. 
Now, when playing the $track, a scanUrl is done and if there are redirections, the 
$track->url is modified to $newurl which will then be used in further getMetadataFor calls.
As this $newurl has no image's link entry in the cache, the "remote_entry_$url" is copied
upon each redirection to "remote_image_newurl" during the scan. Ultimately, when the actual
image is cached, getMetadataFor will get a local link to the image proxy
```

Reply:
Now, maybe we can find a compromise where I add a method named cacheRemoteImage? I'll change the PR